### ISSUE TITLE
My full review, suggestions as well as some quality of life issues I noticed

### ISSUE THREAD

**Issue by soulstenance** [AnySoftKeyboard/AnySoftKeyboard#2461]
First I want to say I love the idea of a FOSS keyboard that doesn't spy on you and I will absolutely be switching when it reach an acceptable level of usability. One thing to note, this app is not visible in the F-Droid app but only shows a long list of language packs instructing you to install the app first. Somewhat annoying being forced to use the Play store.

**Suggestions:**
1. Allow adjusting the size of the keyboard.
2. Let us add a number line to the top.
3. Allow access to common symbols through the main keyboard by long pressing a given key. Each letter key could be assigned a single symbol (in addition to the existing alternate letter forms) that could be activated by long pressing the key.
4. Allow adding, customizing or removing keys on the bottom row. The space bar is tiny right now because of all the extra keys (I'm looking at you, Emoji and apostrophe keys!). The emoji key could be activated by long press of the Enter key or similar.
5. Currently to get suggestions in a given language you need to have that keyboard the active keyboard. It would be nice if (as long as they are using the same writing system), the suggestions would pull from all languages you have installed and once you type a full word, it clues in to what language you are trying to use and actively suggests from that language even if it's not the active keyboard.
6. Swiping space bar to switch keyboards would be a minor but easy QOL change to add.

**Bugs:**

1. You can't exit symbols/numbers without pressing space making it very frustrating the switch back if you didn't want a space after your symbol. This is made worse by the fact this is the ONLY way to type symbols (See suggestion 3 above).
2. If I type a single letter word such as "I" or "a" it auto-corrects to another multi-letter word. Considering how often these words are used this is a huge oversight. It also seems to be impossible to add these single letter words to the dictionary.
3. Some modern devices no long have a physical home or interaction buttons but use gestures instead. The keyboard seems to ignore this, leaving a giant white bar underneath. Very ugly and a waste of space. Minor issue but worth fixing.

**Android OS version:** 10

**Device manufacturer and model:** Samsung Galaxy S9+

**List of installed add-ons (like languages, or themes):** SV language pack

**Comment by Albnu14** [comment]
Hi, I'm not a developer so sorry if this is offensive but your review isn't full,
You can adjust the size of the keyboard and add the number line and change the bottom row (to a certain point) from the settings, they are hidden pretty well and some you wouldn't even think are there which is why I support a redo of the interface ( not that I can do it),
It took me a good long week to know how to do it, I still have hard time searching for some options.
BTW I like your inuyasha picture it got me excited to answer (◕‿◕) .

**Comment by soulstenance** [comment]
> Hi, I'm not a developer so sorry if this is offensive but your review isn't full,
> You can adjust the size of the keyboard and add the number line and change the bottom row (to a certain point) from the settings, they are hidden pretty well and some you wouldn't even think are there which is why I support a redo of the interface ( not that I can do it),
> It took me a good long week to know how to do it, I still have hard time searching for some options.
> BTW I like your inuyasha picture it got me excited to answer (◕‿◕) .

Thanks, I did a little more digging and did find *some* of the settings I wanted. However, the out of box experience is still quite difficult and will turn away non-techsavvy users. I agree with you that the interface needs to be redone. All the settings should be in one place with clearly defined categories rather than in 6 or 8 random hard to find places.

PS: Glad to see Inuyasha fans still exist out in the wild. :smile:

**Comment by gabrc52** [comment]
I agree with that, it took me a while to find the option to reduce the keyboard height, but I don't think it's as easy to design and organize the settings, so we'd need to make a proposal of some sorts

**Comment by Jonta** [comment]
https://github.com/AnySoftKeyboard/AnySoftKeyboard/discussions would be a better place for this, I think =)

**Comment by soulstenance** [comment]
> https://github.com/AnySoftKeyboard/AnySoftKeyboard/discussions would be a better place for this, I think =)

Sorry about that. I figured since there were some bugs listed that it would be good here. But I'll keep that in mind in the future. Thanks. 🙂

**Comment by Jonta** [comment]
Discussions are a pretty new feature. =)

I general, it's easier to handle things like this if they're in separate bugs (in places without the discussion-feature for example), because they can then be crossed off one by one, instead of "ah, 1 of the 9 is still unresolved, can't close this big one yet"

**Comment by soulstenance** [comment]
> Discussions are a pretty new feature. =)
> 
> I general, it's easier to handle things like this if they're in separate bugs (in places without the discussion-feature for example), because they can then be crossed off one by one, instead of "ah, 1 of the 9 is still unresolved, can't close this big one yet"

Oh, that is a cool new feature. And yeah I just had so many issues when I first tried it and I didn't really want to open like 20 tickets. I agree though that would be cleaner. I did cross out some that I later discovered were non-issues. The main thing is the first point in bold, if I'm being honest. More of a preference/suggestion that an actual bug. 🤔🧐😊
